src/dls_bluesky_core/core/scanspecs.py

```diff
@@ -0,0 +1,32 @@
+from typing import List, Optional
+
+import numpy as np
+from scanspec.core import Frames
+from scanspec.specs import DURATION
+
+
+def get_duration(frames: List[Frames]) -> Optional[float]:
+    """
+    Returns the duration of a number of ScanSpec frames, if known and consistent.
+
+    Args:
+        frames (List[Frames]): A number of Frame objects
+
+    Raises:
+        ValueError: If any frames do not have a Duration defined.
+
+    Returns:
+        duration (float): if all frames have a consistent duration
+        None: otherwise
+
+    """
+    for fs in frames:
+        if DURATION in fs.axes():
+            durations = fs.midpoints[DURATION]
+            first_duration = durations[0]
+            if np.all(durations == first_duration):
+                # Constant duration, return it
+                return first_duration
+            else:
+                return None
+    raise ValueError("Duration not specified in Spec")
```

Comment:
Assuming that this is meant to be the duration of one frame, I see nothing wrong with returning an array of durations if there are multiple possible durations. 

I'm also not sure about raising if the duration isn't specified. It's an optional part of the scanspec and the return type here is optional so `return None` feels appropriate.

If you do stick with raising an error, I think `KeyError` is more appropriate since `DURATION` is a key.